### DISCUSSION TITLE
fetch: throwing errors instead of returning them

### DISCUSSION THREAD

**Discussion by RWOverdijk** [openapi-ts/openapi-typescript#1875] [Q&A] [answered]
Is it possible to throw errors instead of returning them? Through some config option for example. Or maybe the way supabase does it with `.throwOnError()`. It would take away some boilerplate, because I now have to do if (error) throw error every time 😄

**Comment by drwpow-figma** [comment, marked as answer]
Yes! There’s a guide [in the docs](https://openapi-ts.dev/openapi-fetch/middleware-auth#throwing) about how to enable this behavior with middleware. By default it won’t throw to match the `fetch` spec.

**Reply by RWOverdijk** [reply]
Figma? Dope!

And my bad, I must've read over it. Silly of me

**Reply by darkbasic** [reply]
@drwpow-figma I've looked at the docs but I get `Property 'error' does not exist on type 'Response'`

```ts
onResponse({ response }) {
  if (response.error) {
    throw new Error(response.error.message);
  }
}
```

![image](https://github.com/user-attachments/assets/84a20921-ab5f-405f-b2ee-876ac2440683)
